src/Flysystem/FlysystemAssetStore.php

```diff
@@ -807,6 +841,7 @@ public function swapPublish($filename, $hash)
 
                 // Blast existing variants from the destination
                 $to->delete($toFileID);
+                $hasher->move($toFileID, $to, '.swap/' . $fromFileID, $from);
```

Comment:
Yeah, the public file needs to be sent back to the protected store, but we can't send it to its final place because there's potentially another file there already.

So we move the file to a `.swap` folder on the protected store instead. Then, we can move our protected file to the public store, which then allows us to move the archived file out of the swap folder.

The hashes of all those files need to follow them around.

In practice, that's not going to make much difference to the file migration because this method should be called during that process. I'm just covering my basis.

Reply:
OK, so if you're batch publishing a lot of files, that can speed up operations (less hashing on intermediaries). If we leave caching disabled outside of file migraiton (my current preference), this doesn't matter - but as you say, it's covering our bases

Reply:
Nothing to do here.